# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# Filename: package/.../iproute2/iproute2-2.6.22-070710-opensde-make.patch
# Copyright (C) 2008 The OpenSDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From d7f4341d8d9137738346252be23c6b45cc9002ea Mon Sep 17 00:00:00 2001
From: Christian Wiese <morfoh@opensde.org>
Date: Tue, 15 Jan 2008 15:38:15 +0200
Subject: [PATCH] Added LIBDIR variable to top-level Makefile


diff --git a/Makefile b/Makefile
index af0d5e4..cc55f2d 100644
--- a/Makefile
+++ b/Makefile
@@ -3,6 +3,7 @@ SBINDIR=/usr/sbin
 CONFDIR=/etc/iproute2
 DOCDIR=/usr/share/doc/iproute2
 MANDIR=/usr/share/man
+LIBDIR=/usr/lib
 
 # Path to db_185.h include
 DBM_INCLUDE:=/usr/include
-- 
1.5.3.5


From cff166a0b0ec53419006ef66f9cba0469711f2b7 Mon Sep 17 00:00:00 2001
From: Christian Wiese <morfoh@opensde.org>
Date: Tue, 15 Jan 2008 15:54:11 +0200
Subject: [PATCH] Improved Makefile to create LIBDIR in install target


diff --git a/Makefile b/Makefile
index cc55f2d..628d3f6 100644
--- a/Makefile
+++ b/Makefile
@@ -42,6 +42,7 @@ Config:
 
 install: all
 	install -m 0755 -d $(DESTDIR)$(SBINDIR)
+	install -m 0755 -d $(DESTDIR)$(LIBDIR)
 	install -m 0755 -d $(DESTDIR)$(CONFDIR)
 	install -m 0755 -d $(DESTDIR)$(DOCDIR)/examples
 	install -m 0755 -d $(DESTDIR)$(DOCDIR)/examples/diffserv
-- 
1.5.3.5


From 1aad4d740909c27962b459a4ff011ba9295f53a5 Mon Sep 17 00:00:00 2001
From: Christian Wiese <morfoh@opensde.org>
Date: Tue, 15 Jan 2008 17:24:05 +0200
Subject: [PATCH] Improved iproute2 to build and install libnetlink.so if SHARED=1 is passed to make


diff --git a/lib/Makefile b/lib/Makefile
index bc270bf..6fe61d1 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -3,7 +3,14 @@ UTILOBJ=utils.o rt_names.o ll_types.o ll_proto.o ll_addr.o inet_proto.o
 
 NLOBJ=ll_map.o libnetlink.o
 
-all: libnetlink.a libutil.a
+NLSHAREDOBJ=
+
+ifeq ($(SHARED),1)
+  CFLAGS += -fPIC -DPIC
+  NLSHAREDOBJ += libnetlink.so
+endif
+
+all: libnetlink.a libutil.a $(NLSHAREDOBJ)
 
 libnetlink.a: $(NLOBJ)
 	$(AR) rcs $@ $(NLOBJ)
@@ -11,8 +18,14 @@ libnetlink.a: $(NLOBJ)
 libutil.a: $(UTILOBJ) $(ADDLIB)
 	$(AR) rcs $@ $(UTILOBJ) $(ADDLIB)
 
+libnetlink.so: $(NLOBJ)
+	$(CC) -shared $(CFLAGS) -o $@ $(NLOBJ)
+
 install:
+ifeq ($(SHARED),1)
+	install -m 0644 $(NLSHAREDOBJ) $(DESTDIR)/$(LIBDIR)
+endif
 
 clean:
-	rm -f $(NLOBJ) $(UTILOBJ) $(ADDLIB) libnetlink.a libutil.a
+	rm -f $(NLOBJ) $(UTILOBJ) $(ADDLIB) libnetlink.a libutil.a $(NLSHAREDOBJ) 
 
-- 
1.5.3.5


From ef1b021137d052028ab30bbd55ef84d89f43bcd4 Mon Sep 17 00:00:00 2001
From: Christian Wiese <morfoh@opensde.org>
Date: Tue, 15 Jan 2008 18:40:21 +0200
Subject: [PATCH] Improved iproute2 to install development headers and libs if DEVEL=1 is passed to make


diff --git a/Makefile b/Makefile
index 628d3f6..ac3f214 100644
--- a/Makefile
+++ b/Makefile
@@ -4,6 +4,7 @@ CONFDIR=/etc/iproute2
 DOCDIR=/usr/share/doc/iproute2
 MANDIR=/usr/share/man
 LIBDIR=/usr/lib
+INCLUDEDIR=/usr/include
 
 # Path to db_185.h include
 DBM_INCLUDE:=/usr/include
@@ -58,6 +59,11 @@ install: all
 	install -m 0755 -d $(DESTDIR)$(MANDIR)/man3
 	install -m 0644 $(shell find man/man3 -maxdepth 1 -type f) $(DESTDIR)$(MANDIR)/man3
 
+ifeq ($(DEVEL),1)
+	install -m 0755 -d $(DESTDIR)$(INCLUDEDIR)
+	install -m 0644 include/{libnetlink.h,ll_map.h} $(DESTDIR)/$(INCLUDEDIR)
+endif
+
 clean:
 	rm -f cscope.*
 	@for i in $(SUBDIRS) doc; \
diff --git a/lib/Makefile b/lib/Makefile
index 6fe61d1..23fa75d 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -26,6 +26,10 @@ ifeq ($(SHARED),1)
 	install -m 0644 $(NLSHAREDOBJ) $(DESTDIR)/$(LIBDIR)
 endif
 
+ifeq ($(DEVEL),1)
+	install -m 0644 libnetlink.a $(DESTDIR)/$(LIBDIR)
+endif
+
 clean:
 	rm -f $(NLOBJ) $(UTILOBJ) $(ADDLIB) libnetlink.a libutil.a $(NLSHAREDOBJ) 
 
-- 
1.5.3.5

